lib/pure/times.nim

```diff
@@ -194,9 +194,14 @@ const
   epochDiff = 116444736000000000'i64
 
 const unitWeights: array[TimeUnit, int64] = [
-  7 * secondsInDay * 1e9.int64, secondsInDay * 1e9.int64,
-  secondsInHour * 1e9.int64, secondsInMin * 1e9.int64,
-  1e9.int64, 1_000_000, 1000, 1
+  1'i64,
+  1000,
+  1_000_000,
+  1e9.int64,
+  secondsInMin * 1e9.int64,
```

Comment:
Is it really needed to use "1e9".int64 instead of 1e9? Just curious, compiler doesn't allow that?

Reply:
Literals using the exponent syntax are always floats.